### PR TITLE
design: PushSecret generator integration

### PR DESCRIPTION
**Pull request by moolen**
Towards: #2643

This PR proposes a direct generator integration into `PushSecret`.

### CODE REVIEW

File: design/009-pushsecret-generator.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+```yaml
+apiVersion: external-secrets.io/v1alpha1
+kind: PushSecret
+spec:
+  selector:
+    generatorRef:
+      apiVersion: generators.external-secrets.io/v1alpha1
+      kind: Password
+      name: "my-password"
```
Comment (shuheiktgw) [inline design/009-pushsecret-generator.md:40]: IIUC, users need to configure `data.match.secretKey` manually to select which secret to push upstream, correct? 🤔 

```
  data:
    - match:
        secretKey: password
        remoteRef:
          remoteKey: some-remote-key
```

Comment (moolen) [reply design/009-pushsecret-generator.md:40]: yes, they have to individually specify the secretKey to select what to push upstream. Once we have PushSecret templates, the user can better define how to structure of the secret.

**Comment by gusfcarvalho** [comment]
should we approve this one formally in the next community meeting? 😄

**Comment by gusfcarvalho** [comment]
Approved in 22nd-november Community Meeting

**Pull request merged**